src/sentry/rules/actions/notify_event_service.py

```diff
@@ -94,17 +94,19 @@ def get_sentry_app_services(self):
 
     def get_plugins(self):
         from sentry.plugins.bases.notify import NotificationPlugin
+        from sentry.plugins.sentry_mail.models import MailPlugin
 
         results = []
         for plugin in plugins.for_project(self.project, version=1):
             if not isinstance(plugin, NotificationPlugin):
                 continue
-            results.append(PluginService(plugin))
+            if self.project.flags.has_issue_alerts_targeting and isinstance(plugin, MailPlugin):
```

Comment:
@jeffkwoh Would it be possible to just override the `is_enabled` function for `MailPlugin` to return `False` if the project has the flag `has_issue_alerts_targeting`?

Reply:
Will look into this/bump this to the person taking over